lib/spack/spack/cmd/buildcache.py

```diff
@@ -249,29 +249,34 @@ def find_matching_specs(pkgs, allow_multiple_matches=False, env=None):
 
     # List of specs that match expressions given via command line
     specs_from_cli = []
-    has_errors = False
-    tty.debug('find_matching_specs: about to parse specs for {0}'.format(pkgs))
-    specs = spack.cmd.parse_specs(pkgs)
+    has_multiple_packages_error = False
+    has_not_installed_error = False
+
     for spec in specs:
         matching = spack.store.db.query(spec, hashes=hashes)
+
         # For each spec provided, make sure it refers to only one package.
         # Fail and ask user to be unambiguous if it doesn't
         if not allow_multiple_matches and len(matching) > 1:
-            tty.error('%s matches multiple installed packages:' % spec)
+            tty.error('{0} matches multiple installed packages:'
+                      .format(spec.format()))
             for match in matching:
                 tty.msg('"%s"' % match.format())
-            has_errors = True
+            has_multiple_packages_error = True
 
         # No installed package matches the query
         if len(matching) == 0 and spec is not any:
-            tty.error('{0} does not match any installed packages.'.format(
-                spec))
-            has_errors = True
+            tty.error('{0} was not installed'.format(spec.format()))
```

Comment:
Why change the error message here?  It could be misconstrued as coming from the install process.

Would the distinction made in the corresponding checking in `cmd/uninstall.py`'s  `find_matching_spec` message be useful?

Reply:
Same as above, let me restore it, it's not important for the bug this pr fixes